src/modelbench/uid.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+import casefy
+
+
+class HasUid:
```

Comment:
I left a few comments about this in [the other PR](https://github.com/mlcommons/modelbench/pull/386#issuecomment-2197726110)!